map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,218 @@
+      mapView.setMap(map);
+
+      // set the map views's viewpoint centered on Los Angeles, California and scaled
+      mapView.setViewpoint(new Viewpoint(new Point(-13185535.98, 4037766.28,
+        SpatialReferences.getWebMercator()), 7000));
+
+      // create a button that toggles the location tracking
+      Button trackingButton = new Button("Start tracking");
+      trackingButton.setDisable(true);
+
+      // create a graphics overlay for the points and use a red circle for the symbols
+      GraphicsOverlay locationHistoryOverlay = new GraphicsOverlay();
+      SimpleMarkerSymbol locationSymbol = new SimpleMarkerSymbol(
+        SimpleMarkerSymbol.Style.CIRCLE, 0xFFFF0000, 10f);
+      SimpleRenderer locationHistoryRenderer = new SimpleRenderer(locationSymbol);
+      locationHistoryOverlay.setRenderer(locationHistoryRenderer);
+
+      // create a graphics overlay for the lines connecting the points and use a green line for the symbol
+      GraphicsOverlay locationHistoryLineOverlay = new GraphicsOverlay();
+      SimpleLineSymbol locationLineSymbol = new SimpleLineSymbol(
+        SimpleLineSymbol.Style.SOLID, 0xFF00FF00, 2.0f);
+      SimpleRenderer locationHistoryLineRenderer = new SimpleRenderer(locationLineSymbol);
+      locationHistoryLineOverlay.setRenderer(locationHistoryLineRenderer);
+
+      // add the graphics overlays to the map view
+      mapView.getGraphicsOverlays().addAll(Arrays.asList(locationHistoryOverlay, locationHistoryLineOverlay));
+
+      // create a polyline builder to connect the location points
+      PolylineBuilder polylineBuilder = new PolylineBuilder(SpatialReferences.getWebMercator());
+
+      try {
+        // access the json of the location points
+        String polylineData = IOUtils.toString(getClass().getResourceAsStream("/show_location_history/polyline_data.json"),
+          StandardCharsets.UTF_8);
+        // create a polyline from the location points
+        Polyline routePolyline = (Polyline) Geometry.fromJson(polylineData,
+          SpatialReferences.getWebMercator());
+
+        // create a simulated location data source
+        SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+        // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+        simulatedLocationDataSource.setLocations(
+          routePolyline, new SimulationParameters(Calendar.getInstance(), 30.0, 0.0, 0.0));
+
+        // configure the map view's location display to follow the simulated location data source
+        LocationDisplay locationDisplay = mapView.getLocationDisplay();
+        locationDisplay.setLocationDataSource(simulatedLocationDataSource);
+        locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+        locationDisplay.setInitialZoomScale(7000);
+
+        // track previous location to ensure the route line appears behind the location display symbol
```

Comment:
Maybe we can be a bit more descriptive about what this block does. The second part of the sentence sounds a bit like we're admitting it's a workaround, so perhaps we can reword it slightly to say this is deliberate?

```suggestion
        // create a listener to track the previous location, to draw a route line behind the location display symbol
```